tests/TestHarness/Node.py

```diff
@@ -550,6 +550,38 @@ def findInLog(self, searchStr):
                         return True
         return False
 
+    # verify only one or two 'Starting block' per block number unless block is restarted
+    def verifyStartingBlockMessages(self):
+        dataDir=Utils.getNodeDataDir(self.nodeId)
+        files=Node.findStderrFiles(dataDir)
+        for f in files:
+            blockNumbers = set()
+            duplicateBlockNumbers = set()
+            duplicatesFound = False
+            lastRestartBlockNum = 0
+
+            with open(f, 'r') as file:
+                for line in file:
+                    match = re.match(r".*Restarting exhausted speculative block #(\d+)", line)
+                    if match:
+                        lastRestartBlockNum = match.group(1)
+                    if re.match(r".*unlinkable_block_exception", line):
+                        lastRestartBlockNum = blockNumber
```

Comment:
`blockNumber` is first defined in line 572. Are we sure it has already been defined when `if re.match(r".*unlinkable_block_exception", line):` holds?